### ISSUE TITLE
Support for insertion ignore for Duplicate Key Entry 

### ISSUE THREAD

**Issue by jmai00** [typeorm/typeorm#143]
Hi,
Is there a way to do this at the moment?

**Comment by pleerock** [comment]
What exactly behaviour should be when entity is inserted with duplicate key entry? I think `INSERT...ON DUPLICATE KEY UPDATE` maybe useful sometimes, however I can't imagine how it can fit ORM context. Because if you call `persist` method and send it entity with id that already exist in the database, ORM will never do `INSERT` again - it will use `UPDATE` since entity with such id is already in the db.

**Comment by jmai00** [comment]
Here's my example situation:
I have many workstations and users. Any given time snapshot, the data is save into the DB. I don't want duplicate users entry.

`import {User} from "./User";
import {Column, JoinColumn, OneToOne, Table, PrimaryGeneratedColumn} from "typeorm";

@Table("Workstation")
export class Workstation {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({name: "name", type: "string"})
    name: string;

    @OneToOne((type) => User, {cascadeInsert: true, nullable: false})
    @JoinColumn({name: "userId"})
    user: User;
}`


`import {Table, Column, PrimaryGeneratedColumn, Index} from "typeorm";

@Table()
@Index("index_first_last_name", (user: User) => [user.firstName, user.lastName], {unique: true})
export class User {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    firstName: string;

    @Column()
    lastName: string;
}`

`require("reflect-metadata");
import {createConnection, ConnectionOptions, Connection} from "typeorm";
import {Workstation} from "./entities/Workstation";
import {User} from "./entities/User";

let dbConnection = {
    type: "mysql",
    host: "localhost",
    port: 3306,
    username: "root",
    password: "root",
    database: "test_dev",
};

createConnection({
    driver: dbConnection,
    entities: [
        __dirname + "/entities/*.js",
    ],
    dropSchemaOnConnection: true,
    autoSchemaSync: true,
} as ConnectionOptions).then( async (connection: Connection) => {
    let x = connection.getRepository(Workstation);

    let user1 = new User();
    user1.firstName = "Hello";
    user1.lastName = "World";

    let workstation1 = new Workstation();
    workstation1.name = "Windows7";
    workstation1.user =  user1;

    let workstation2 = new Workstation();
    workstation2.name = "Linux";
    workstation2.user = user1;

    let user2 = new User();
    user2.firstName = user1.firstName;
    user2.lastName = user1.lastName;

    let workstation3 = new Workstation();
    workstation3.name = "Mac";
    workstation3.user = user2;

    await x.persist(workstation1);
    await x.persist(workstation2);
    await x.persist(workstation3);
    console.log("Done!");

}).catch( (error) => {
    console.log(`Error: ${error.message}`);
});`


In the case above:
workstation1 and workstation2 has the same user object: user1.
workstation3 has a different user object but the same value as user1

I get the error:
>Error: ER_DUP_ENTRY: Duplicate entry 'Hello-World' for key 'index_first_last_name'

**Comment by pleerock** [comment]
how many users do you want to save? If only one, then why do you create two users, instead of using same user everywhere?

p.s. please make sure to format your code correctly ;)

**Comment by jmai00** [comment]
One user, but I'm trying to simulate 2 different records, as that is what my situation is. The record workstation comes in at different times but the user information is the same.

**Comment by pleerock** [comment]
you need to loop up a db and check if you dont have a user yet, only after that create a new user. otherwise use old one.

```typescript

    let user1 = new User();
    user1.firstName = "Hello";
    user1.lastName = "World";

    let workstation1 = new Workstation();
    workstation1.name = "Windows7";
    workstation1.user =  user1;

    let workstation2 = new Workstation();
    workstation2.name = "Linux";
    workstation2.user = user1;

    await x.persist(workstation1);
    await x.persist(workstation2);


    let user2 = await connection.getRepository(User).findOneById();
    if (!user2) { 
         const user2 = new User();
         user2.firstName = user1.firstName;
         user2.lastName = user1.lastName;
    }

    let workstation3 = new Workstation();
    workstation3.name = "Mac";
    workstation3.user = user2;

    await x.persist(workstation3);
    console.log("Done!");
```

**Comment by dimiork** [comment]
Here is behaviour that needed to use INSERT IGNORE:
 The user selects data for the week and loads them into the application.
The application parse data and saves to MySQL table.
Another user selects data for the month, which also contains data for the week that are already written in the table.
Unfortunately, there is no any any identificators in data, so we need to create multiple-column index to prevent duplicate data in table.
The number of data uploaded by users will be considered large enough to make a check of availability for each item in the table. 
So, application make a batch insert using QueryBuilder.
"INSERT IGNORE" is what I see the only way to avoid duplicates in the table.
How can I implement this case?
May be it is possible to stringify QueryBuilder's generated SQL-query where inserted data is already escaped and then pass it to getRepository.query() ?
However, even it is possible it's a dirtiest ever huck. Can you provide any correct solution? Thx a lot.

**Comment by dimiork** [comment]
@pleerock , pay attention to my message  above, please

**Comment by pleerock** [comment]
@dimiork I can suggest you two things:

1. use raw query. Its not hard to create a simple `INSER INTO ... IGNORE ...` query and you don't need to use QueryBuilder. For special cases like yours its okay to have exceptions and 
2. create a detailed proposal in a separate issue on how you see this is going to be implemented in typeorm and implement this feature for all exist drivers. Its not complex task, 2 of 5 I would say.
